stable/redis/templates/metrics-prometheus.yaml

```diff
@@ -0,0 +1,26 @@
+{{- if and (.Values.metrics.enabled) (.Values.metrics.serviceMonitor.enabled) }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "redis.fullname" . }}
+  {{- if .Values.metrics.serviceMonitor.namespace }}
+  namespace: {{ .Values.metrics.serviceMonitor.namespace }}
+  {{- end }}
+  labels:
+    app: {{ template "redis.name" . }}
+    chart: {{ template "redis.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    {{- range $key, $value := .Values.metrics.serviceMonitor.selector }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  endpoints:
+  - interval: {{ .Values.metrics.serviceMonitor.interval }}
+    port: metrics
+  selector:
+    matchLabels:
+      app: {{ template "redis.name" . }}
+  namespaceSelector:
+      any: true
```

Comment:
You should be aware of the intention to deprecate this feature. Looks like ServiceMonitors will only be able to select services in the same namespace 

https://github.com/coreos/prometheus-operator/issues/2122

Reply:
Understood, thanks for the heads up.  For now, we deploy into the same namespace, and this is just a pattern I've seen in other places, e.g. https://github.com/helm/charts/blob/master/incubator/kafka/templates/servicemonitors.yaml#L21